diff -ru bochs-2.4.6.orif/configure bochs-2.4.6/configure
--- bochs-2.4.6.orif/configure	2012-02-04 22:15:35.406192999 +0100
+++ bochs-2.4.6/configure	2012-02-04 22:16:12.799323757 +0100
@@ -22040,7 +22040,7 @@
 	  		PCIDEV_MODULE_MAKE_ALL="all-kernel24"
 			KERNEL_MODULE_SUFFIX="o"
           	;;
-		2.6*)
+		2.6*|3.0*)
 	  		PCIDEV_MODULE_MAKE_ALL="all-kernel26"
 			KERNEL_MODULE_SUFFIX="ko"
 		;;
