diff -ru grub-2.00.orig/util/grub-mkfont.c grub-2.00/util/grub-mkfont.c
--- grub-2.00.orig/util/grub-mkfont.c	2016-08-30 15:08:15.194978661 +0200
+++ grub-2.00/util/grub-mkfont.c	2016-08-30 15:08:43.764286004 +0200
@@ -39,7 +39,7 @@
 #include FT_FREETYPE_H
 #include FT_TRUETYPE_TAGS_H
 #include FT_TRUETYPE_TABLES_H
-#include <freetype/ftsynth.h>
+#include FT_SYNTHESIS_H
 
 #undef __FTERRORS_H__
 #define FT_ERROR_START_LIST   const char *ft_errmsgs[] = { 
